cassovary-core/src/test/scala/com/twitter/cassovary/util/io/ListOfEdgesGraphReaderSpec.scala

```diff
@@ -79,19 +82,95 @@ class ListOfEdgesGraphReaderSpec extends WordSpec with GraphBehaviours[Node] wit
     }
 
     "using String ids" should {
+
+      def hasCorrectProperties(graph:  SharedArrayBasedDirectedGraph): Unit = {
+        graph.nodeCount should be(6)
+        graph.edgeCount should be(6L)
+        graph.maxNodeId should be(5)
+      }
+
       "provide the correct graph properties" in {
-        new GraphWithStringIds {
-          graph.nodeCount should be(6)
-          graph.edgeCount should be(6L)
-          graph.maxNodeId should be(5)
+        new GraphWithStringIds("toy_6nodes_list_StringIds.txt") {
+          hasCorrectProperties(graph)
         }
       }
       "contain the right nodes and edges" in {
-        new GraphWithStringIds {
+        new GraphWithStringIds("toy_6nodes_list_StringIds.txt") {
+          behave like renumberedGraphEquivalentToMap(graph, stringGraphMap, seqNumberer)
+        }
+      }
+
+      "provide the correct graph properties with sorting reader" in {
+        new GraphWithStringIds("toy_6nodes_list_StringIds.txt", sortNeighbors = true) {
+          hasCorrectProperties(graph)
+        }
+      }
+      "contain the right nodes and edges with sorting reader" in {
+        new GraphWithStringIds("toy_6nodes_list_StringIds.txt", sortNeighbors = true) {
+          behave like renumberedGraphEquivalentToMap(graph, stringGraphMap, seqNumberer)
+        }
+      }
+
+      "provide the correct graph properties with duplicate elimination reader" in {
+        new GraphWithStringIds("toy_6nodes_list_StringIds.txt", removeDup = true) {
+          hasCorrectProperties(graph)
+        }
+      }
+      "contain the right nodes and edges with duplicate elimination reader" in {
+        new GraphWithStringIds("toy_6nodes_list_StringIds.txt", removeDup = true) {
           behave like renumberedGraphEquivalentToMap(graph, stringGraphMap, seqNumberer)
         }
       }
+
+      "provide the correct graph properties with sorting and duplicate elimination reader" in {
+        new GraphWithStringIds("toy_6nodes_list_StringIds.txt", sortNeighbors = true, removeDup = true) {
+          hasCorrectProperties(graph)
+        }
+      }
+      "contain the right nodes and edges with sorting and duplicate elimination reader" in {
+        new GraphWithStringIds("toy_6nodes_list_StringIds.txt", sortNeighbors = true, removeDup = true) {
+          behave like renumberedGraphEquivalentToMap(graph, stringGraphMap, seqNumberer)
+        }
+      }
+
+      "on file with duplicated edges provide the correct graph properties with duplicate elimination reader" in {
+        new GraphWithStringIds("toy_6nodes_list_StringIds_dup_edges.txt", removeDup = true) {
+          hasCorrectProperties(graph)
+        }
+      }
+      "on file with duplicated edges contain the right nodes and edges with duplicate elimination reader" in {
```

Comment:
Frankly speaking I tried :-) I kept one empty line when the graph/reader changes and no empty lines for different tests of the same graph.